pkg/sync/adapter_test.go

```diff
@@ -97,16 +99,17 @@ func TestSyncFiles(t *testing.T) {
 	}
 
 	ctrl := gomock.NewController(t)
+	kc := kclient.NewMockClientInterface(ctrl)
+	kc.EXPECT().ExecCMDInContainer(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).
```

Comment:
I was looking at your test in your PR. Not an expert of gomock, but to some extent, yes, I think you could have used `gomock.Any()` to mock interactions with `o.kubernetesClient`. However, I think this would be of limited help given the signature of the `matchOwnerReferenceWithResources` method you are testing: you would still need to construct and pass `resources []unstructured.Unstructured` to it, which seems to be the point of your `fakePod` and `fakeDeployment` functions.